code/game/objects/items/shuttle_creator.dm

```diff
@@ -0,0 +1,301 @@
+		stationary_port.Destroy()
+		linkedShuttleId = null
+		return FALSE
+	port.dir = invertedDir
+	port.port_direction = portDirection
+
+	if(!calculate_bounds(port))
+		to_chat(usr, "<span class='warning'>Bluespace calculations failed, please select a new airlock.</span>")
+		port.Destroy()
+		stationary_port.Destroy()
+		linkedShuttleId = null
+		return FALSE
+
+	port.shuttle_areas = list()
+	//var/list/all_turfs = port.return_ordered_turfs(port.x, port.y, port.z, port.dir)
+	var/list/all_turfs = loggedTurfs
+	for(var/i in 1 to all_turfs.len)
+		var/turf/curT = all_turfs[i]
+		var/area/cur_area = curT.loc
+		//Add the area to the shuttle <3
+		if(istype(cur_area, recorded_shuttle_area))
+			if(istype(curT, /turf/open/space))
+				continue
+			if(length(curT.baseturfs) < 2)
+				continue
+			//Add the shuttle base shit to the shuttle
+			curT.baseturfs.Insert(3, /turf/baseturf_skipover/shuttle)
+			port.shuttle_areas[cur_area] = TRUE
+
+	port.linkup(new_shuttle, stationary_port)
+
+	port.movement_force = list("KNOCKDOWN" = 0, "THROW" = 0)
+	port.initiate_docking(stationary_port)
+
+	port.mode = SHUTTLE_IDLE
+	port.timer = 0
+
+	port.register()
+	GLOB.custom_shuttle_count ++
+	message_admins("[ADMIN_LOOKUPFLW(user)] created a new shuttle with a [src] at [ADMIN_VERBOSEJMP(user)] ([GLOB.custom_shuttle_count] custom shuttles, limit is [CUSTOM_SHUTTLE_LIMIT])")
+	log_game("[key_name(user)] created a new shuttle with a [src] at [AREACOORD(user)] ([GLOB.custom_shuttle_count] custom shuttles, limit is [CUSTOM_SHUTTLE_LIMIT])")
+	return TRUE
+
+/obj/item/shuttle_creator/proc/create_shuttle_area(mob/user)
+	if(!loggedTurfs)
+		return FALSE
+	//Create the new area
+	var/area/shuttle/custom/powered/newS
+	var/area/oldA = loggedOldArea
+	var/str = stripped_input(user, "Shuttle Name:", "Blueprint Editing", "", MAX_NAME_LEN)
+	if(!str || !length(str))
+		return FALSE
+	if(length(str) > 50)
+		to_chat(user, "<span class='warning'>The provided ship name is too long, blares the [src]</span>")
+		return FALSE
+	if(OOC_FILTER_CHECK(str))
+		to_chat(user, "<span class='warning'>Nanotrasen prohibited words are in use in this shuttle name, blars the [src] in a slightly offended tone.</span>")
+		return FALSE
+	newS = new /area/shuttle/custom/powered()
+	newS.setup(str)
+	newS.set_dynamic_lighting()
+	//Shuttles always have gravity
+	newS.has_gravity = TRUE
+	newS.requires_power = TRUE
+	//Record the area for use when creating the docking port
+	recorded_shuttle_area = newS
+
+	for(var/i in 1 to loggedTurfs.len)
+		var/turf/turf_holder = loggedTurfs[i]
+		var/area/old_area = turf_holder.loc
+		newS.contents += turf_holder
+		turf_holder.change_area(old_area, newS)
+
+	newS.reg_in_areas_in_z()
+
+	var/list/firedoors = oldA.firedoors
+	for(var/door in firedoors)
+		var/obj/machinery/door/firedoor/FD = door
+		FD.CalculateAffectingAreas()
+	return TRUE
+
+/obj/item/shuttle_creator/proc/check_current_area(mob/user)
+	var/static/area_or_turf_fail_types = typecacheof(list(
```

Comment:
Why is this static?

Reply:
Because the code is a modified version of /code/__HELPERS/areas.dm /proc/create_area(mob/creator) and it was copied across